examples/medplum-photon-integration/src/bots/sync-patient.ts

```diff
@@ -157,6 +179,135 @@ async function updatePatient(medplum: MedplumClient, patient: Patient, result: a
   }
 }
 
+export async function formatPhotonInput(
+  patient: Patient,
+  medplum: MedplumClient,
+  authToken: string,
+  inputType: 'AllergyIntolerance' | 'MedicationRequest'
+): Promise<(PhotonAllergenInput | PhotonMedHistoryInput)[] | undefined> {
+  const inputs: (PhotonAllergenInput | PhotonMedHistoryInput)[] = [];
+  const resources = await medplum.searchResources(inputType, {
+    patient: getReferenceString(patient),
+  });
+
+  if (resources.length === 0) {
+    return undefined;
+  }
+
+  for (const resource of resources) {
+    const photonId = await getPhotonId(resource, authToken);
+    if (!photonId) {
+      continue;
+    }
+
+    const input = createInput(resource, photonId);
+    inputs.push(input);
+  }
+
+  return inputs;
+}
+
+function createInput(
+  resource: AllergyIntolerance | MedicationRequest,
+  photonId: string
+): PhotonAllergenInput | PhotonMedHistoryInput {
+  if (resource.resourceType === 'AllergyIntolerance') {
+    const input: PhotonAllergenInput = {
+      allergenId: photonId,
+      comment: resource.note?.[0].text,
+    };
+    if (resource.onsetDateTime) {
+      input.onset = formatAWSDate(resource.onsetDateTime);
+    }
+
+    return input;
+  } else {
+    const input: PhotonMedHistoryInput = {
+      medicationId: photonId,
+      active: resource.status === 'active',
+      comment: resource.note?.[0].text,
+    };
+    return input;
+  }
+}
+
+async function getPhotonId(
```

Comment:
Having a bunch of `? : ` that switch on the resource type is messy. You should factor out the common parts into a helper function to avoid code duplication, and then create an Allergy specific / Meds specific functions that call this helper